pandas/tests/frame/test_alter_axes.py

```diff
@@ -908,3 +908,37 @@ def test_set_reset_index(self):
         df = df.set_index('B')
 
         df = df.reset_index()
+
+    def test_set_axis(self):
+
+        df = DataFrame({'A': [1.1, 2.2, 3.3],
+                        'B': [5.0, 6.1, 7.2],
+                        'C': [4.4, 5.5, 6.6]},
+                       index=[2010, 2011, 2012])
+
+        expected = {0: df.copy(),
+                    1: df.copy()}
+        expected[0].index = list('abc')
+        expected[1].columns = list('abc')
+        expected['index'] = expected[0]
+        expected['columns'] = expected[1]
+
+        for axis in expected:
+            # inplace=True
+            for inplace, warn in (None, FutureWarning), (True, None):
+                kwargs = {'inplace': inplace}
+
+                result = df.copy()
+                with tm.assert_produces_warning(warn):
+                    result.set_axis(axis, list('abc'), **kwargs)
+                tm.assert_frame_equal(result, expected[axis])
+
+            # GH14636 (inplace=False)
```

Comment:
Are you sure? The scope of the test is broader (since ``set_axis()`` wasn't tested at all) than just the issue, which specifically refers to ``inplace=False``.